baseplate/sidecars/live_data_watcher.py

```diff
@@ -120,6 +68,20 @@ def on_change(self, data: Optional[bytes], _znode_stat: ZnodeStat) -> None:
             self.handle_empty_data()
             return
 
+        loader_type = _parse_loader_type(data)
+        if loader_type == LoaderType.S3:
+            try:
+                data = _load_from_s3(data)
+            except LoaderException:
+                logger.error("Failed to load data from S3. Not writing to destination file")
+                return
+
+        if data is None:
+            logger.warning(
+                "No data to write to destination file. Something is likely misconfigured."
+            )
+            return
```

Comment:
would prematurely returning here cause an application to potentially fail if the file doesn't exist? Should it be an error (maybe tracked with a span/metric of some sort that can alert on an error, sentry would suffice if it's monitored) and self-correct to an empty string to write?